tests/functional_small/rom/lspg_unsteady/main10.cc

```diff
@@ -0,0 +1,136 @@
+
+#include <gtest/gtest.h>
+#include "pressio/rom_subspaces.hpp"
+#include "pressio/rom_lspg_unsteady.hpp"
+
+namespace{
+
+template<class phi_t>
+struct FakeNonLinSolver
+{
+  int N_ = {};
+  phi_t phi_;
+  double dt_;
+
+  FakeNonLinSolver(int N, phi_t phi, double dt)
+    : N_(N), phi_(phi), dt_(dt){}
+
+  template<class SystemType, class StateType>
+  void solve(const SystemType & system, StateType & romState)
+  {
+    auto R = system.createResidual();
+    auto J = system.createJacobian();
+
+    // ensure that we call the system multiples times to simulate multiple
+    // iterations of a nonlinear solver
+    for (int k=0; k<11; ++k){
+      system.residualAndJacobian(romState, R, &J);
+      for (int i=0; i<romState.size(); ++i){ romState(i) += 1.; }
+    }
+    // std::cout << "R = \n" << R << std::endl;
+    // std::cout << "J = \n" << J << std::endl;
+  }
+};
+
+class MyFom
+{
+  using phi_type = Eigen::MatrixXd;
+  int N_ = {};
+  mutable int hookCount_ = 0;
+
+public:
+  using time_type = double;
+  using state_type = Eigen::VectorXd;
+  using discrete_residual_type = state_type;
+
+  explicit MyFom(int N) : N_(N){}
+
+  discrete_residual_type createDiscreteTimeResidual() const{
+    return discrete_residual_type(N_);
+  }
+
+  phi_type createResultOfDiscreteTimeJacobianActionOn(const phi_type & B) const{
+    return phi_type(N_, B.cols());
+  }
+
+  template<class StepIntType>
+  void preStepHook(StepIntType sc, double time, double dt) const
+  {
+    // this should be called only once per time step
+    std::cout << "hook: " << sc << " " << time << " " << dt << "\n";
```

Comment:
This `std::cout` call could be replaced with the logger:

```suggestion
    PRESSIOLOG_INFO("hook: {}, time: {}\n", sc, dt);
```